test/lib/interpreters/split_clustering_test.rb

```diff
@@ -948,6 +948,27 @@ def test_list_vehicles
       # if schedule is less than a week, we generate one cluster per vehicle per day
       assert_equal 5, Interpreters::SplitClustering.list_vehicles({ start: 0, end: 4 }, vrp.vehicles, :work_day).size
     end
+
+    def test_split_with_vehicle_alternative_skills
+      vrp = TestHelper.create(VRP.lat_lon_scheduling_two_vehicles)
+
+      # nothing expected to be raised :
+      [[[]], [['skill']], [['skill'], []]].each{ |skill_set|
+        vrp.vehicles.first.skills = skill_set
+        Interpreters::SplitClustering.split_balanced_kmeans(
+          { vrp: vrp, service: :demo }, vrp.vehicles.size,
+          ntity: :vehicle, restarts: 1, max_iterations: 1)
+      }
```

Comment:
This is a general comment: it is dangerous to modify the VRP object directly like this in the tests (unless it is necessary or the test is about that) because we skip the validations done in coerce and create function. 

Changes in these functions will not be represented in the tests and we risk having a test that will not fail when it should or vice-versa, a test that breaks when it really shouldn't. 

The safe option is to always modify the json and call the create on the modified problem when it is necessary 
```suggestion
      problem = VRP.lat_lon_scheduling_two_vehicles

      # nothing expected to be raised :
      [[[]], [['skill']], [['skill'], []]].each{ |skill_set|
        problem[:vehicles].first[:skills] = skill_set
        Interpreters::SplitClustering.split_balanced_kmeans(
          { vrp: TestHelper.create(problem), service: :demo }, problem[:vehicles].size,
          entity: :vehicle, restarts: 1, max_iterations: 1)
      }
```